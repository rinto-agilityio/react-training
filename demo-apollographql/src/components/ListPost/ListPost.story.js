import React from 'react'
import { storiesOf } from '@storybook/react'
import { MemoryRouter } from 'react-router-dom'

// Components
import ListPost from './ListPost'

// Mock data
import posts from './__mocks/db'

storiesOf('List', module)
  .add('ListPost', () => (
    <MemoryRouter>
      <ListPost posts={posts} />
    </MemoryRouter>
  ))
