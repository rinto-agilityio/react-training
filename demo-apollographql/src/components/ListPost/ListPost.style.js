import styled from 'styled-components'

// Components
import Paper from '@material-ui/core/Paper'
import { Link } from 'react-router-dom'

export const ItemWrapper = styled(Paper)({
  padding: '1rem',
  marginBottom: '1rem'
})

export const ItemLink = styled(Link)({
  color: 'inherit',
  textDecoration: 'none',
})
