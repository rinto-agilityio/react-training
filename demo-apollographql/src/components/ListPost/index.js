import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Link } from 'react-router-dom'

const ListPostWrapper = styled.section`
  display: grid;
  grid-template-rows: auto;
  grid-row-gap: 20px;

`
const SinglePostWrapper = styled.article`
  background-color: #fff;
  padding: 20px;
  border: 1px solid rgba(0,0,0,.09);
  border-radius: 3px;
  box-shadow: 0 1px 4px rgba(0,0,0,.03);
`
const StyledLink = styled(Link)`
  color: inherit;
  text-decoration: none;
`

const ListPost = ({ posts }) => (
  <ListPostWrapper>
    {posts.map(post => (
      <SinglePostWrapper key={post.id}>
        <StyledLink to={`post/${post.slug}`} title={post.title}>
          <h2>{post.title}</h2>
          <p>{post.content}</p>
        </StyledLink>
      </SinglePostWrapper>
    ))}
  </ListPostWrapper>
)

ListPost.propTypes = {
  posts: PropTypes.array.isRequired
}

export default ListPost
