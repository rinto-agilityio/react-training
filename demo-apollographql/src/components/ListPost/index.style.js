import styled from 'styled-components'
import { Link } from 'react-router-dom'

const ListPostWrapper = styled.section`
  display: grid;
  grid-template-rows: auto;
  grid-row-gap: 20px;

`
const SinglePostWrapper = styled.article`
  background-color: #fff;
  padding: 20px;
  border: 1px solid rgba(0,0,0,.09);
  border-radius: 3px;
  box-shadow: 0 1px 4px rgba(0,0,0,.03);
`
const StyledLink = styled(Link)`
  color: inherit;
  text-decoration: none;
`

export {
  ListPostWrapper,
  SinglePostWrapper,
  StyledLink,
}
