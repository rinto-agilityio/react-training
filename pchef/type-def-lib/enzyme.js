declare module 'enzyme' {
  declare module.exports: any;
}
