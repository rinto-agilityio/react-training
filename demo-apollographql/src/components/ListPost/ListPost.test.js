import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

// Components
import ListPost from './index'
import { ItemWrapper } from './ListPost.style'

// Mock data
import posts from './__mocks/db'

describe('Components', () => {
  describe('<ListPost />', () => {
    it('Snapshot renders correctly', () => {
      const tree = renderer
        .create(
          <MemoryRouter>
            <ListPost posts={posts} />
          </MemoryRouter>
        )
        .toJSON()

        expect(tree).toMatchSnapshot()
    })

    it(`Render ${posts.length} child components`, () => {
      const wrapper = shallow(<ListPost posts={posts} />)

      expect(wrapper.find(ItemWrapper).length).toEqual(posts.length)
    })
  })
})
