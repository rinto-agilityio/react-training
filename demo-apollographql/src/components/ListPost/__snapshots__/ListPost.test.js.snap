// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components <ListPost /> Snapshot renders correctly 1`] = `
Array [
  <div
    className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 sc-bdVaJa fjlkyF"
  >
    <a
      className="sc-bwzfXH eczfXd"
      href="/post/mocking-es-and-commonjs-modules-with-jest-mock"
      onClick={[Function]}
      title="Mocking ES and CommonJS modules with jest.mock()"
    >
      <h2>
        Mocking ES and CommonJS modules with jest.mock()
      </h2>
      <p>
        In order for a unit test to be worthwhile it must be reliable; it should not be reliant on hopeful consistency of external dependencies in order to pass. External dependencies can include any modules/packages external to the test subject, requests to a database, or requests to an API. If running the same test several times without change can result in different outcomes, the test loses value. If trust in the test is lost, less focus is inevitably paid to test failures.
      </p>
    </a>
  </div>,
  <div
    className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 sc-bdVaJa fjlkyF"
  >
    <a
      className="sc-bwzfXH eczfXd"
      href="/post/making-sense-of-react-hooks"
      onClick={[Function]}
      title="Making Sense of React Hooks"
    >
      <h2>
        Making Sense of React Hooks
      </h2>
      <p>
        We know that components and top-down data flow help us organize a large UI into small, independent, reusable pieces. However, we often can’t break complex components down any further because the logic is stateful and can’t be extracted to a function or another component. Sometimes that’s what people mean when they say React doesn’t let them “separate concerns.”
      </p>
    </a>
  </div>,
  <div
    className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 sc-bdVaJa fjlkyF"
  >
    <a
      className="sc-bwzfXH eczfXd"
      href="/post/why-side-projects-are-so-damn-important"
      onClick={[Function]}
      title="Why side projects are so damn important"
    >
      <h2>
        Why side projects are so damn important
      </h2>
      <p>
        Learning and mastering a new skill can be tough, but with enough persistence and motivation there is a lot we can achieve. Like learning how to code, it can be hard, frustrating and time-consuming. In the end, the most important thing to do is to keep going and gain experience.
      </p>
    </a>
  </div>,
  <div
    className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 sc-bdVaJa fjlkyF"
  >
    <a
      className="sc-bwzfXH eczfXd"
      href="/post/progressive-enhancement-with-css-grid"
      onClick={[Function]}
      title="Progressive enhancement with CSS Grid"
    >
      <h2>
        Progressive enhancement with CSS Grid
      </h2>
      <p>
        CSS Grid (Grid) has been out for some time now, with full support in major modern browsers. I’ll leave others to dive into why it’s so great and what new design possibilities it makes so easy to explore. If you have been looking for the best way to support responsive web designs, I’ve yet to see anyone that doesn’t love Grid. It’s simple to use, yet extremely powerful and flexible.
      </p>
    </a>
  </div>,
  <div
    className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 sc-bdVaJa fjlkyF"
  >
    <a
      className="sc-bwzfXH eczfXd"
      href="/post/react-components-elements-and-instances"
      onClick={[Function]}
      title="React Components, Elements, and Instances"
    >
      <h2>
        React Components, Elements, and Instances
      </h2>
      <p>
        Many people get confused by the difference between components, their instances, and elements in React. Why are there three different terms to refer to something that is painted on screen?
If you’re new to React, you probably only worked with component classes and instances before. For example, you may declare a Button component by creating a class. When the program is running, you may have several instances of this component on screen, each with its own properties and local state. This is the traditional object oriented UI programming. Why introduce elements?
      </p>
    </a>
  </div>,
  <div
    className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 sc-bdVaJa fjlkyF"
  >
    <a
      className="sc-bwzfXH eczfXd"
      href="/post/the-evolution-of-flux-frameworks"
      onClick={[Function]}
      title="The Evolution of Flux Frameworks"
    >
      <h2>
        The Evolution of Flux Frameworks
      </h2>
      <p>
        There has been no shortage of great Flux implementations, such as Flummox, Alt, or Fluxible. Most of them are focused on making Flux easier to use with the server rendering and reducing the boilerplate. They also often provide convenience utilities like higher-order components and asynchronous action helpers. Still, under the hood, many of them are built on top of the original Flux Dispatcher.
      </p>
    </a>
  </div>,
  <div
    className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 sc-bdVaJa fjlkyF"
  >
    <a
      className="sc-bwzfXH eczfXd"
      href="/post/how-to-create-and-publish-a-chrome-extension-in-20-minutes"
      onClick={[Function]}
      title="How to Create and Publish a Chrome Extension in 20 minutes"
    >
      <h2>
        How to Create and Publish a Chrome Extension in 20 minutes
      </h2>
      <p>
        Ever wondered what it would be like to create a Chrome extension? Well, I’m here to tell you just how easy it is. Follow these steps and your idea will turn into reality and you’ll be able to publish a real extension in the Chrome Web Store in no time.
      </p>
    </a>
  </div>,
  <div
    className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 sc-bdVaJa fjlkyF"
  >
    <a
      className="sc-bwzfXH eczfXd"
      href="/post/you-might-not-need-redux"
      onClick={[Function]}
      title="You Might Not Need Redux"
    >
      <h2>
        You Might Not Need Redux
      </h2>
      <p>
        People often choose Redux before they need it. “What if our app doesn’t scale without it?” Later, developers frown at the indirection Redux introduced to their code. “Why do I have to touch three files to get a simple feature working?” Why indeed!
      </p>
    </a>
  </div>,
  <div
    className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 sc-bdVaJa fjlkyF"
  >
    <a
      className="sc-bwzfXH eczfXd"
      href="/post/what-are-encryption-keys-and-how-do-they-work"
      onClick={[Function]}
      title="What are encryption keys and how do they work?"
    >
      <h2>
        What are encryption keys and how do they work?
      </h2>
      <p>
        In cryptography a ‘key’ is a piece of information used in combination with an algorithm (a ‘cipher’) to transform plaintext into ciphertext (encryption) and vice versa (decryption).
A cipher can be ‘reciprocal’ if it is used for both encryption and decryption, or ‘non-reciprocal’ if a transformation to the key is required when using it in reverse.
      </p>
    </a>
  </div>,
  <div
    className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 sc-bdVaJa fjlkyF"
  >
    <a
      className="sc-bwzfXH eczfXd"
      href="/post/here-s-what-i-learned-at-the-world-s-biggest-react-conference"
      onClick={[Function]}
      title="Here’s what I learned at the world’s biggest React conference"
    >
      <h2>
        Here’s what I learned at the world’s biggest React conference
      </h2>
      <p>
        In her talk, Tracy showed why reactive programming can be a more efficient way to code. She also discussed how it has been adopted by industry leaders such as Netflix, Slack, Microsoft, and Facebook as the new standard for developing applications. It seems very promising, especially libraries like RxJS, that help developers deliver complex features quicker with less, more maintainable code.
      </p>
    </a>
  </div>,
]
`;
